Watchman.Discord/Areas/Administration/Controllers/AdministrationController.cs

```diff
@@ -87,34 +87,74 @@ public async Task ReadUserMessages(DiscordRequest request, Contexts contexts)
         [DiscordCommand("set role")]
         public async Task SetRoleAsSafe(DiscordRequest request, Contexts contexts)
         {
-            var args = request.Arguments.Skip(1).ToArray(); // 1 args is string "role", so it's not needed
-            if (args.Length < 2)
+            var args = request.Arguments.Skip(1).ToList(); // 1 args is string "role", so it's not needed
+
+            if (args.Count() < 2)
             {
                 throw new NotEnoughArgumentsException();
             }
 
-            var roleName = args[0].Value;
-            var toSetAsSafe = args[1].Value == "safe";
-
-            var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
-            if (serverRole == null)
+            if (args.HasDuplicates())
             {
-                throw new RoleNotFoundException(roleName);
+                throw new ArgumentsDuplicatedException();
             }
 
-            if (toSetAsSafe)
+            var isSafeArgument = args.Last().Value.ToLower();
+
+            if (isSafeArgument != "safe"
+                && isSafeArgument != "unsafe")
             {
-                var command = new SetRoleAsSafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                throw new NotEnoughArgumentsException();
             }
-            else
+
+            var isSafe = isSafeArgument == "safe" ? true : false;
+
+            var safeRoles = this._queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles;
+            var argRoleNames = args.Select(x => x.Value).SkipLast(1).ToList();
+
+            foreach (var roleNameToCheck in argRoleNames)
             {
-                var command = new SetRoleAsUnsafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                var serverRole = _usersRolesService.GetRoleByName(roleNameToCheck, contexts.Server);
+
+                if (serverRole == null)
+                {
+                    throw new RoleNotFoundException(roleNameToCheck);
```

Comment:
hmm tutaj zrobiłbym żeby informowało o tym użytkownika, a nie wyrzucało wyjątek 

powinno powinformować i spróbować dodać kolejną, żeby było przyjemniejsze w użyciu 
jak ktoś poda 5 roli
to chciałby żeby 4 mu dodało, a tą z tą jedną najwyżej poprawi później

Reply:
wyjątek miałby sens gdyby dało sie dodawać max jedną role jednocześnie 
ale w przypadku dodawania wielu, lepiej podejść do tego sposobem "zróbny tyle ile sie da"